### ISSUE TITLE
Open a directory recursively

### ISSUE THREAD

**Issue by TornaxO7** [kyazdani42/nvim-tree.lua#380]
Is it possible to open a directory recursively for example when pressing `O`? I'd like to see my project structure with nvim-tree.

**Comment by kyazdani42** [comment]
not yet, it could be implemented though, but kind of slow depending on the numbers of directories under the directory you try to open

**Comment by TornaxO7** [comment]
Ok, should I close this issue for you?

**Comment by kyazdani42** [comment]
nope, i will forget otherwise ^^

**Comment by sarahkittyy** [comment]
Please implement this, I switched from NERDTree to this and this is the one feature i miss the most. ❤️

**Comment by 0x00002a** [comment]
I created a function to do this based on `collapse-all.lua`:
```lua
function ExpandAll()
    local nvimlib = require('nvim-tree.lib')
    local function iter(nodes)
        for _, node in pairs(nodes) do
            if not node.open and node.nodes then
                nvimlib.expand_or_collapse(node)
                --node.open = true
            end
            if node.nodes then
                iter(node.nodes)
            end
        end
    end

    local nodes = {}
    local currnode = nvimlib.get_node_at_cursor()
    if currnode == nil then
        return
    end
    nodes = {currnode}
    iter(nodes)
end
```
it will expand all nodes under the current cursor, probably not optimal but it works :p

**Comment by kyazdani42** [comment]
implemented in #1292
